src/snap/invokeSnap.ts

```diff
@@ -0,0 +1,31 @@
+import { Page } from "puppeteer";
+import { SnapInstallationParamNames } from "../types";
+import { flaskOnly } from "./utils";
+
+export async function invokeSnap(
+  page: Page,
+  snapId: string,
+  method: string,
+  params: Record<SnapInstallationParamNames, any> = {}
+): Promise<Partial<any>> {
+  flaskOnly(page);
+  return page.evaluate(
+    async (opts: {
+      snapId: string;
+      method: string;
+      params: Record<SnapInstallationParamNames, any>;
+    }) => {
+      return window.ethereum.request({
+        method: "wallet_invokeSnap",
+        params: [
+          `${opts.snapId}`,
+          {
+            method: opts.method,
+            ...opts.params,
```

Comment:
invoke snap accepts `SnapRequest` interface, so params should be under params

```suggestion
            params: opts.params,
```